### DISCUSSION TITLE
Slice is not rendered on the server side

### DISCUSSION THREAD

**Discussion by karelvrabeckv** [Kitware/trame#476] [Trame/VTK] [answered]
Hello,
I have implemented **a slice tool** using `vtkExtractVOI()`.

It works perfectly on the **client** side (`vtkLocalView`):
![client](https://github.com/Kitware/trame/assets/62759688/d4e55088-7e73-4e97-bdb2-c1cb1ddd2aa9)

But whenever I switch to the **server** side (`vtkRemoteView`) or look into the server window, I see only this:
![server](https://github.com/Kitware/trame/assets/62759688/80ece8df-93a1-4bbc-a7cd-04e3e490cf23)

The work process for creating the slice is as follows:

```
    def create_slice(self, file: File):
        slice = vtkExtractVOI()
        
        image_data = file.reader.GetOutput()
        slice.SetInputData(image_data)
        
        return slice

    def create_slice_mapper(self, file: File, slice, group_active_array, lookup_table):
        slice_mapper = vtkPolyDataMapper()
        
        slice_mapper.SetInputConnection(slice.GetOutputPort())
        slice_mapper.SetScalarRange(file.data.GetArray(group_active_array).GetRange())
        slice_mapper.SetLookupTable(lookup_table)
        
        return slice_mapper

    def create_slice_actor(self, slice_mapper):
        slice_actor = vtkActor()

        slice_actor.SetMapper(slice_mapper)
        slice_actor.GetProperty().LightingOff()
        
        return slice_actor
```

I just create **a slice**, **mapper**, and **actor** and call `SetVOI(...)` for setting the orientation and position of the slice.

Could anyone please tell me what I am doing wrong?

Thank you very much! :)

**Comment by jourdain** [comment]
That is a first... Usually the remote rendering one should be the one to just work unless you are doing something wrong on the VTK side. The fact that is working with the local view is intriguing. 

Could you provide a reproducible Python script similar to what you can find [here](https://examples.vtk.org/site/Python/)?

**Reply by karelvrabeckv** [reply, marked as answer]
This was my mistake. I used `vtkPolyDataMapper()` instead of `vtkDataSetMapper()`. Now, it works without a problem.
